### PR TITLE
Rename transform to rewrite

### LINKED ISSUE #169: Rename the command `transform` to `rewrite`
## Subtasks

- [ ] Rename the CLI command `transform` to `rewrite`.
- [ ] Update docs on the site.

### PR DESCRIPTION
**Pull request by RoukayaZaki**
- closes #169

---

## PR-Codex overview
This PR focuses on renaming the transformation functionality in the `eo-phi-normalizer` project to "rewrite," reflecting a change in terminology and functionality. It updates documentation, command-line interface, and internal references accordingly.

### Detailed summary
- Updated documentation to replace `transform` with `rewrite`.
- Changed command-line interface from `CLI'TransformPhi` to `CLI'RewritePhi`.
- Modified parser and command names to reflect the new "rewrite" terminology.
- Adjusted references in the `CHANGELOG.md` and other documentation files.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/docs/src/pipeline.md [hunk 1]
```diff
@@ -46,11 +46,11 @@ The pipeline has several stages:
 
 ### Modify pipeline configuration
 
-The `pipeline/config.yaml` file specifies how original `EO` programs are transformed into initial `EO` programs.
-Transformation is necessary because some objects in original `EO` programs don't work.
+The `pipeline/config.yaml` file specifies how original `EO` programs are rewriteed into initial `EO` programs.
+Rewriteation is necessary because some objects in original `EO` programs don't work.
```
Comment (deemp) [inline site/docs/src/pipeline.md:50]: All these (funny) changes in this file should be removed.

Comment (RoukayaZaki) [reply site/docs/src/pipeline.md]: sorry for that

Comment (deemp) [reply site/docs/src/pipeline.md]: No problem. Just remove all changes to that file.

File: site/docs/src/pipeline.md [hunk 2]
```diff
@@ -46,11 +46,11 @@ The pipeline has several stages:
 
 ### Modify pipeline configuration
 
-The `pipeline/config.yaml` file specifies how original `EO` programs are rewriteed into initial `EO` programs.
-Rewriteation is necessary because some objects in original `EO` programs don't work.
+The `pipeline/config.yaml` file specifies how original `EO` programs are rewrote into initial `EO` programs.
+Rewriting is necessary because some objects in original `EO` programs don't work.
```
Comment (deemp) [inline site/docs/src/pipeline.md:50]: These changes should be removed too.

I suggest you:
- Undo the commits https://github.com/objectionary/normalizer/pull/369/commits/45064e68e9a1f18c57fa97fdc397de14066168ad, https://github.com/objectionary/normalizer/pull/369/commits/85a053b7fca3f7fba892c87a735e42d96570bc70
- Unstage changes in this file
- Commit again
- Force push. Force push is safe on a PR branch.

**Comment by deemp** [comment]
Additionally, I'd suggest you:
- Use Conventional Commits ([link](https://www.conventionalcommits.org/en/v1.0.0/))
- Don't apologize because
  - "It's never personal, you're just doing your job" ([link](https://www.reddit.com/r/cscareerquestions/comments/5ya7e3/comment/deoskoc/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)).
  - This is not the `master` branch and nobody was hurt.
  - I think it's more productive to write a message that asks for clarification or that outlines how you're going to fix the problem than writing a message where you apologize for creating this problem.
  - [link](https://www.huffpost.com/entry/dont-apologize-say-thank-you_b_10758972)

**Review by aabounegm — APPROVED** [review]

**Comment by deemp** [comment]
@RoukayaZaki, please resolve the conflicts

File: eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs [hunk 3]
```diff
@@ -305,11 +305,11 @@ msplit (Chain m) = Chain $ \ctx ->
     [] -> runChain (return Nothing) ctx
     (logs, x) : xs -> [(logs, Just (x, Chain (const xs)))]
 
-transformLogs :: (log1 -> log2) -> Chain log1 a -> Chain log2 a
```
Comment (deemp) [inline eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs:308]: Keep `transformLogs`

File: eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs [hunk 4]
```diff
@@ -305,11 +305,11 @@ msplit (Chain m) = Chain $ \ctx ->
...
-transformLogs f (Chain normChain) = Chain $ map (first (map (fmap f))) . normChain
+rewriteLogs :: (log1 -> log2) -> Chain log1 a -> Chain log2 a
+rewriteLogs f (Chain normChain) = Chain $ map (first (map (fmap f))) . normChain
 
-transformNormLogs :: NormalizeChain a -> DataizeChain a
```
Comment (deemp) [inline eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs:311]: @RoukayaZaki, keep `transformNormLogs`

File: site/docs/src/normalizer.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Normalizer
```
Comment (deemp) [inline site/docs/src/normalizer.md:1]: Why are there new files?

File: site/docs/src/pipeline.md [hunk 6]
```diff
@@ -58,7 +58,7 @@ Make `cygpath` available on `PATH`.
 
 The pipeline has several important stages:
 
-- Transform original `EO` programs into filtered `EO` programs, leaving only the specified top-level objects (tests).
+- rewrite original `EO` programs into filtered `EO` programs, leaving only the specified top-level objects (tests).
```
Comment (deemp) [inline site/docs/src/pipeline.md:61]: Capitalize: "Rewrite"

File: eo-phi-normalizer/src/Language/EO/Phi/Dataize.hs [hunk 7]
```diff
@@ -116,7 +116,7 @@ dataizeRecursivelyChain = fmap minimizeObject' . go
               logStep "Normalized" obj'
               return obj'
           | otherwise = applyRulesChainWith limits obj
-    msplit (transformNormLogs normalizedObj) >>= \case
```
Comment (deemp) [inline eo-phi-normalizer/src/Language/EO/Phi/Dataize.hs:119]: @RoukayaZaki, keep `transformNormLogs`

**Pull request merged**